waqi_client_async/__init__.py

```diff
@@ -29,30 +33,51 @@ class OverQuota(APIError):
     pass
 
 
+class UnknownCity(APIError):
+    """Raised when the provided city could not be found."""
+
+    pass
+
+
+class UnknownID(APIError):
+    """Raised when the provided ID could not be found."""
+
+    pass
+
+
 class UnknownStation(APIError):
     """Raised when the provided station could not be found."""
 
     pass
 
 
 def assert_valid(result: dict) -> None:
-    if result.get("status") != "error":
-        return
+    LOGGER.debug("JSON Data: %s", result)
+    if result.get("status") == "ok":
+        if data := result.get("data") is not None:
+            if data.get("msg") == "Unknown ID":
+                raise UnknownID()
 
-    data = result.get("data")
-    if data == "Invalid key":
+    return
+
+    message = result.get("msg")
+    if message == "Invalid key":
         raise InvalidToken()
-    elif data == "Over quota":
+    elif message == "Over quota":
         raise OverQuota()
-    elif data == "Unknown station":
-        raise UnknownStation()
-    elif data:
-        raise APIError(data)
-
-    message = result.get("message")
-    if message:
+    elif message == "Unknown city":
+        raise UnknownCity()
+    elif message:
         raise APIError(message)
 
+    keyword = result.get("keyword")
+    if keyword == "no_matching_stations_found":
+        raise APIError(keyword)
+
+    data = result.get("data")
+    if data == "no such station":
+        raise UnknownStation()
+
```

Comment:
Enabling this code by removing the above `return`, I have been trying, without success, to reproduce the potential JSON result that would match those updated checks. Could you please share the URLs for which it reproduced? I would want to write `pytest` tests for that.

Reply:
I have captured some results that has lead me to write these code-snippets.

But, as I can't see the full data-packets that are returned, I'm not sure if they are correct.

I have added a LOGGER.debug-line with the purpose to be able to see the full data-packets that are returned.
With this complete info I had planned to make the error-handling code final and fully working.

So, please release a version 1.0.1 based on the present, corrected code (see my last commit below) and allow me to capture more data, that I happily will share with you. And let's then create a v1.1.0 with correct error-handling and added testcases.